lib/syskit/exceptions.rb

```diff
@@ -447,38 +449,50 @@ def initialize(device, task0, task1, toplevel_tasks_to_requirements = {})
             @device = device
             @tasks = [task0, task1]
 
-            solver = NetworkGeneration::MergeSolver.new(task0.plan)
-            @merge_result = solver.resolve_merge(task0, task1, {})
             @involved_definitions = @tasks.map do |t|
                 find_all_related_syskit_actions(t, toplevel_tasks_to_requirements)
             end
         end
 
-        def find_all_related_syskit_actions(task, toplevel_tasks_to_requirements)
-            result = []
-            while task
-                result.concat(toplevel_tasks_to_requirements[task] || [])
-                task = task.each_parent_task.first
-            end
-            result
-        end
-
         def pretty_print(pp)
             pp.text "device '#{device.name}' of type #{device.model} is assigned "
             pp.text "to two tasks that cannot be merged"
-            pp.breakable
-            @merge_result.pretty_print_failure(pp)
-            @involved_definitions.each_with_index do |defs, i|
-                next if defs.empty?
+            print_failed_merge_chain(pp, *@tasks)
+            @tasks.zip(@involved_definitions).each do |t, defs|
+                print_dependent_definitions(pp, t, defs)
+            end
+        end
+    end
 
-                pp.breakable
-                pp.text "Chain #{i + 1} is needed by the following definitions:"
-                pp.nest(2) do
-                    defs.each do |d|
-                        pp.breakable
-                        pp.text d.to_s
-                    end
+    class ConflictingDeploymentAllocation < SpecError
+        include Syskit::NetworkGenerationsExceptionHelpers
+
+        attr_reader :deployment_to_tasks, :toplevel_tasks_to_requirements
+
+        def initialize(deployment_to_tasks, toplevel_tasks_to_requirements = {})
+            @deployment_to_tasks = deployment_to_tasks
+            @toplevel_tasks_to_requirements = toplevel_tasks_to_requirements
+        end
+
+        def pretty_print(pp)
+            deployment_to_tasks.each do |orocos_name, tasks|
+                agent = tasks.first.execution_agent
+                deployment_m = agent.deployed_orogen_model_by_name(orocos_name)
+                pp.text(
+                    "deployed task '#{orocos_name}' from deployment " \
+                    "'#{deployment_m.name}' defined in " \
+                    "'#{deployment_m.project.name}' on '#{agent.process_server_name}' " \
+                    "is assigned to #{tasks.size} tasks. Bellow is the list of " \
+                    "the dependent non-deployed actions. Right after the list is " \
```

Comment:
```suggestion
                    "the dependent non-deployed actions. Right after the list " \
```

Sorry about the split reviews...I am figuring out how to adapt this message to be used in the context of the error capture feature, as it is task-centered, meaning that I have to individualize the error for each involved task.